tests/wamr-test-suites/spec-test-script/runtest.py

```diff
@@ -1122,11 +1135,24 @@ def test_assert_with_exception(form, wast_tempfile, wasm_tempfile, aot_tempfile,
                     # workaround: spec test changes error message to "malformed" while iwasm still use "invalid"
                     error_msg = m.group(2).replace("malformed", "invalid")
                     log("Testing(malformed)")
-                    f = file(wasm_tempfile, 'w')
+                    f = open(wasm_tempfile, 'wb' if IS_PY_3 else "w")
+                    # f = open(wasm_tempfile, 'w')
                     s = m.group(1)
+                    print("==>s: {}".format(s))
                     while s:
                         res = re.match("[^\"]*\"([^\"]*)\"(.*)", s, re.DOTALL)
-                        f.write(res.group(1).replace("\\", "\\x").decode("string_escape"))
+                        binary_module = res.group(1)
+
+                        print("==>1. SZ:{} binary_module: \"{}\"".format(len(binary_module), binary_module))
+                        if IS_PY_3:
+                            binary_module = binary_module.replace("\\", "\\x").encode("raw_unicode_escape").decode("unicode_escape")
+                            binary_module = binary_module.encode("latin-1")
```

Comment:
maybe it's simpler to use codecs.escape_decode
```
import codecs

b = binary_module.replace("\\", "\\x")
(binary_module, _) = codecs.escape_decode(b)
```

Reply:
thanks. It works. 
But I can not make sure it is a standard function of codecs because I just can't locate `codecs.escape_decode` in [python3.7](https://docs.python.org/3.7/library/codecs.html), [python 3.8](https://docs.python.org/fr/3.8/library/codecs.html), [python 3.10](https://docs.python.org/3.10/library/codecs.html#module-codecs). But I did try `"escape_decode" in dir(codecs)` and got a `True`. 

Would you mind helping to make sure of it?

Reply:
it seems that it's undocumented. i didn't know that.
maybe it's better to stick with documented methods.